src/components/MultiHopTrade/hooks/useAllowanceApproval/helpers.ts

```diff
@@ -0,0 +1,107 @@
+import { fromAssetId } from '@shapeshiftoss/caip'
+import type { evm, EvmChainAdapter } from '@shapeshiftoss/chain-adapters'
+import type { ETHWallet, HDWallet } from '@shapeshiftoss/hdwallet-core'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { bn } from 'lib/bignumber/bignumber'
+import type { TradeQuote } from 'lib/swapper/api'
+import { erc20AllowanceAbi } from 'lib/swapper/swappers/utils/abi/erc20Allowance-abi'
+import { MAX_ALLOWANCE } from 'lib/swapper/swappers/utils/constants'
+import {
+  getApproveContractData,
+  getERC20Allowance,
+  getFeesFromContractData,
+} from 'lib/swapper/swappers/utils/helpers/helpers'
+import { getWeb3InstanceByChainId } from 'lib/web3-instance'
+
+export const checkApprovalNeeded = async (
+  tradeQuoteStep: TradeQuote['steps'][number],
+  wallet: HDWallet,
+) => {
+  const { sellAsset, accountNumber, allowanceContract } = tradeQuoteStep
+  const adapterManager = getChainAdapterManager()
+  const adapter = adapterManager.get(sellAsset.chainId)
+
+  if (!adapter) throw Error(`no chain adapter found for chain Id: ${sellAsset.chainId}`)
+  if (!wallet) throw new Error('no wallet available')
+
+  // No approval needed for selling a fee asset
+  if (sellAsset.assetId === adapter.getFeeAssetId()) {
+    return false
+  }
+
+  const from = await adapter.getAddress({
+    wallet,
+    accountNumber,
+  })
+
+  const { assetReference: sellAssetContractAddress } = fromAssetId(sellAsset.assetId)
+  const web3 = getWeb3InstanceByChainId(sellAsset.chainId)
+
+  const allowanceOnChainCryptoBaseUnit = await getERC20Allowance({
+    web3,
+    erc20AllowanceAbi,
+    address: sellAssetContractAddress,
+    spender: allowanceContract,
+    from,
+  })
+
+  // TODO(woodenfurniture): This was pulled from the old implementation but we should check whether we should be
+  // including protocol fees in the sell asset here
```

Comment:
I don't believe so, allowance should be for a specific amount, regardless of this being within the context of a trade or not. If we think this may fail with multi-hops, we can always add a buffer

Reply:
This is a good point. So long as `sellAmountBeforeFeesCryptoBaseUnit` always refers to the actual amount being sent to the contract address then nothing needs to change. 

Looking at all current swappers I think that is the case, though if so, the variable naming might be a little misleading, as it implies there might be a different (higher) amount _actually_ leaving the wallet from the sell asset.

I'm not sure of a better name though, as the current name is not incorrect.